### PR TITLE
chore(maven): update the maven wrapper

### PR DESCRIPTION
**Pull request by ckipp01**
Looks like this can potentially cause issues with Maven builds that are
requiring a more modern version of Maven.

refs #4023

### CODE REVIEW

**Comment by tgodzik** [comment]
Seems we are getting:
` no main manifest attribute,`
but we are running via `-jar` so maybe something changed and we need to add the main class?

**Comment by ckipp01** [comment]
> Seems we are getting: ` no main manifest attribute,` but we are running via `-jar` so maybe something changed and we need to add the main class?

Yup, I'll look into it a bit tonight. Also the wrapper we are using is actually deprecated and maven publishes their own wrapper now, so I'm looking at using that instead.

**Comment by tgodzik** [comment]
> Maybe you remember from before @tgodzik but the old `maven-wrapper.jar` that we have does have the `Main-class` specified in the manifest, however when I pull that same version down and look, it's not there:
> 
> ```
> cp $(cs fetch io.takari:maven-wrapper:0.5.5) test.jar
> ```
> 
> And then when I look at the manifest, there is no main. Do you remember how you got this jar? I am able to get it to work by instead using `-cp <jar> <main>` which I _think_ should be fine, especially since we are testing this, but it is a bit different.

I think I just downloaded it from github. Or maybe I also modified it to have the main class in the manifest? That was a long time ago and I am not super sure about it. Are we able to migrate to the official wrapper?

**Comment by ckipp01** [comment]
> Are we able to migrate to the official wrapper?

Yup, tests are now ✅ .

File: .jvmopts [hunk 1]
```diff
@@ -1,6 +1,5 @@
 -Xss4m
--Xms1G
--Xmx2G
+-Xmx1G
```
Comment (tgodzik) [inline .jvmopts:2]: Do we need to change `.jvmopts`? Did you find that these were not neccessary? I think I changed these for the CI :thinking:

**Review by tgodzik — APPROVED** [review]
LGTM! Just one question otherwise

File: .jvmopts (hunk 1 above)
Comment (ckipp01) [reply .jvmopts:2]: Ach, sorry, I was running `.bin/test.sh` which changes these automatically. I will revert this.

Comment (tgodzik) [reply .jvmopts]: Ach, so `-Xmx1G` is actually used for the CI, totally forgot

**Pull request merged**